lib/mongoid/errors/mongoid_error.rb

```diff
@@ -48,7 +48,7 @@ def compose_message(key, attributes = {})
       #
       # @return [ String ] A localized error message string.
       def translate(key, options)
-        ::I18n.translate("#{BASE_KEY}.#{key}", options)
+        ::I18n.translate("#{BASE_KEY}.#{key}", **options)
```

Comment:
My thinking with Ruby 3 support is that the double-splat syntax should be used everywhere for options.

Therefore, for example, this method's signature should be changed to take **options, and so on up the call stack.

Reply:
@p-mongo lets do that as a separate PR. The focus of this PR should be to get Ruby 3.0 to pass the tests with the minimal amount of code changes.

Reply:
We generally leave it up to the commenters to resolve conversations on their comments. This helps with the review process as the responses are visible.

I generally try not to add hacks to Mongoid (or driver, etc.) because later on they end up causing larger problems. Ruby 3 uses the double splat syntax for options, therefore it is my expectation that a Ruby 3-compatible release of Mongoid would do the same unless there is a clear reason not to.

Reply:
Sure, you're welcome to add this for the "official Ruby 3" release. I have renamed this PR as "Preliminary Ruby 3.0 support" in order to clarify that I am simply getting the tests to pass, so that people who want to use an alpha version on Ruby 3.0 don't have to fork the official repo. I will leave it up to the MongoDB team to add whatever they feel is needed for official Ruby 3.0 support before they publicly announce it. Sound fair?

Reply:
Resolving this as per above comment.